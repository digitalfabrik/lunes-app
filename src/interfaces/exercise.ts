export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
