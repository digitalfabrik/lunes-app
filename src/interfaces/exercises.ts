export interface IExercisesScreenProps {
  route: any
  navigation: any
}
