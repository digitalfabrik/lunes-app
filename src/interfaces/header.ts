export interface IHeaderProps {
  top: number | undefined
}
