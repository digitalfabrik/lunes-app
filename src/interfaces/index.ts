import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
