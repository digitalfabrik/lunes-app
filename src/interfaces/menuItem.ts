import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
