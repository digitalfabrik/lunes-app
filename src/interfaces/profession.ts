export interface IProfessionScreenProps {
  navigation: any
}

export interface IProfessionsProps {
  id: number
  title: string
  description: string
  icon?: string
  total_training_sets: number
}
