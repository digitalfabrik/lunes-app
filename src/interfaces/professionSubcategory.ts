export interface IProfessionSubcategoryScreenProps {
  route: any
  navigation: any
}

export interface IProfessionSubcategoryProps {
  id: number
  title: string
  description: string
  icon: string
  total_documents: number
}
