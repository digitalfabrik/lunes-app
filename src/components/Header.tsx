import React from 'react'
import { View, StyleSheet } from 'react-native'
import { SquareIcon, StarIcon, CircleIcon, LinesIcon, SmileIcon } from '../../assets/images'
import { IHeaderProps } from '../interfaces/header'
import { COLORS } from '../constants/colors'

export const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: COLORS.lunesBlack
  },
  header: {
    backgroundColor: COLORS.lunesBlack,
    height: 91,
    width: '100%'
  },
  squareIcon: {
    position: 'absolute',
    top: 41.5,
    left: 24.3,
    width: 28,
    height: 28
  },
  starIcon: {
    position: 'absolute',
    top: -5,
    left: 98.3,
    width: 28,
    height: 23.5
  },
  circleIcon: {
    position: 'absolute',
    top: 37.5,
    right: 68.8,
    width: 28,
    height: 29
  },
  verticalLinesIcon: {
    position: 'absolute',
    top: 16.5,
    right: 0,
    width: 24.5,
    height: 28
  },
  smileIcon: {
    position: 'absolute',
    width: 80,
    height: 80,
    left: '50%',
    marginLeft: -40,
    top: 51
  }
})
const Header = ({ top }: IHeaderProps) => (
  <View style={{ ...styles.wrapper, paddingTop: top }} testID='header'>
    <View style={styles.header}>
      <View style={styles.squareIcon}>
        <SquareIcon />
      </View>
      <View style={styles.starIcon}>
        <StarIcon />
      </View>
      <View style={styles.circleIcon}>
        <CircleIcon />
      </View>
      <View style={styles.verticalLinesIcon}>
        <LinesIcon />
      </View>
      <View style={styles.smileIcon}>
        <SmileIcon />
      </View>
    </View>
  </View>
)

export default Header
