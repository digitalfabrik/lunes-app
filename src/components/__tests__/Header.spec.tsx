import 'react-native'
import React from 'react'
import Header from '../Header'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { IHeaderProps } from '../../interfaces/header'

describe('Components', () => {
  describe('Header', () => {
    const defaultHeaderProps: IHeaderProps = {
      top: 5
    }

    it('renders correctly across screens', () => {
      const component = shallow(<Header {...defaultHeaderProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should have top padding = top prop passed to it', () => {
      const headerProps: IHeaderProps = {
        ...defaultHeaderProps
      }
      const paddingTop = 5

      const component = shallow(<Header {...headerProps} />)
      expect(component.find('[testID="header"]').props().style).toHaveProperty('paddingTop', paddingTop)
    })
  })
})
