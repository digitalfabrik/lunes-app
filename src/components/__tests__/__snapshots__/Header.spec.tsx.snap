// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header renders correctly across screens 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000e38",
      "paddingTop": 5,
    }
  }
  testID="header"
>
  <View
    style={
      Object {
        "backgroundColor": "#000e38",
        "height": 91,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 28,
          "left": 24.3,
          "position": "absolute",
          "top": 41.5,
          "width": 28,
        }
      }
    >
      <[object Object] />
    </View>
    <View
      style={
        Object {
          "height": 23.5,
          "left": 98.3,
          "position": "absolute",
          "top": -5,
          "width": 28,
        }
      }
    >
      <[object Object] />
    </View>
    <View
      style={
        Object {
          "height": 29,
          "position": "absolute",
          "right": 68.8,
          "top": 37.5,
          "width": 28,
        }
      }
    >
      <[object Object] />
    </View>
    <View
      style={
        Object {
          "height": 28,
          "position": "absolute",
          "right": 0,
          "top": 16.5,
          "width": 24.5,
        }
      }
    >
      <[object Object] />
    </View>
    <View
      style={
        Object {
          "height": 80,
          "left": "50%",
          "marginLeft": -40,
          "position": "absolute",
          "top": 51,
          "width": 80,
        }
      }
    >
      <[object Object] />
    </View>
  </View>
</View>
`;
